### PR TITLE
Bound Tenure Expand

### PR DESCRIPTION
**Pull request by amicic**
Fixing issue in https://github.com/eclipse/omr/issues/3492

When running Gencon with Concurrent Mark disabled, prevent the unbounded
growth of Tenure space.

Two new percolate reasons are introduced:
- if Tenure free heap was already high (at least midway between min and
max bounds) in previous global GC, and it's suspected that Scavenge will
expand Tenure even more (due to low current Tenure free, and based on
historic tenuring activity) start global GC instead
- if Tenure free heap was not very high in previous GC, we'll allow
Scavenges and Tenure expansion, but we will break the streak when
hitting the max free bound, and start global GC

Signed-off-by: Aleksandar Micic <amicic@ca.ibm.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
@dmitripivkine can you review this as well please?

**Comment by amicic** [comment]
added a couple of trace points

File: gc/base/standard/Scavenger.cpp [hunk 1]
```diff
@@ -4192,6 +4192,31 @@ MM_Scavenger::internalGarbageCollect(MM_EnvironmentBase *envBase, MM_MemorySubSp
...
 	}
 
+	if (!_extensions->concurrentMark) {
+			uintptr_t previousUsedOldHeap = _extensions->oldHeapSizeOnLastGlobalGC - _extensions->freeOldHeapSizeOnLastGlobalGC;
+			float maxTenureFreeRatio = _extensions->heapFreeMaximumRatioMultiplier / 100.0f;
+			float midTenureFreeRatio = (_extensions->heapFreeMinimumRatioMultiplier + _extensions->heapFreeMaximumRatioMultiplier) / 200.0f;
+			uintptr_t soaFreeMemorySize = _extensions->heap->getApproximateActiveFreeMemorySize(MEMORY_TYPE_OLD) - _extensions->heap->getApproximateActiveFreeLOAMemorySize(MEMORY_TYPE_OLD);
+
+			/* We suspect that next scavegne will cause Tenure expansion, while Tenure free ratio is (was) already high enough */
+			if ((scavengerGCStats->_avgTenureBytes > soaFreeMemorySize)	&& (_extensions->freeOldHeapSizeOnLastGlobalGC > (_extensions->oldHeapSizeOnLastGlobalGC * midTenureFreeRatio))) {
```
Comment (charliegracie) [inline gc/base/standard/Scavenger.cpp]: there is some weird spacing on this line

**Comment by charliegracie** [comment]
@genie-omr build all

File: gc/base/GCExtensionsBase.hpp [hunk 2]
```diff
@@ -1419,7 +1426,6 @@ class MM_GCExtensionsBase : public MM_BaseVirtual {
 		, concurrentScavenger(false)
 		, concurrentScavengerForced(false)
 		, concurrentScavengerHWSupport(false)
-		, concurrentScavengerBackgroundThreads(1)
```
Comment (amicic) [inline gc/base/GCExtensionsBase.hpp]: just noticed that I mistakenly removed a line here. must fix this first

**Comment by charliegracie** [comment]
@genie-omr build all

**Pull request merged**